framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -139,54 +129,96 @@ private static Pattern anchored(String regex) {
-            return parseIntLiteral(expression, types);
-        } else if (isLongLiteral(expression, context)) {
-            return parseLongLiteral(expression, types);
-        } else if (isFloatLiteral(expression, context)) {
-            throw constructParserException(
-                    expression,
-                    String.format("floating-point values '%s' cannot be parsed", expression));
-        } else if (isStringLiteral(expression, context)) {
-            return parseStringLiteral(expression, types, env.getElementUtils());
-        } else if (isThisLiteral(expression, context)) {
-            return parseThis(context);
-        } else if (isSuperLiteral(expression, context)) {
-            return parseSuper(expression, types, context);
-        } else if (isIdentifier(expression)) {
-            return parseIdentifier(expression, env, path, context);
-        } else if (isParameter(expression, context)) {
-            return parseParameter(expression, context);
-        } else if (isArray(expression)) {
-            return parseArray(expression, context, path);
-        } else if (isMethodCall(expression)) {
-            return parseMethodCall(expression, context, path, env);
-        } else if (isMemberSelect(expression)) {
-            return parseMemberSelect(expression, env, context, path);
-        } else if (isParentheses(expression, context)) {
-            return parseParentheses(expression, context, path);
-        } else {
-            String message;
-            if (expression.equals("#0")) {
-                message =
-                        "one should use \"this\" for the receiver or \"#1\" for the first formal parameter";
+        if (expr != null) {
+            if (expr.isNullLiteralExpr()) {
+                return new ValueLiteral(types.getNullType(), (Object) null);
+            } else if (expr.isIntegerLiteralExpr() && !context.parsingMember) {
+                return new ValueLiteral(
+                        types.getPrimitiveType(TypeKind.INT), expr.asIntegerLiteralExpr().asInt());
+            } else if (expr.isLongLiteralExpr() && !context.parsingMember) {
+                return new ValueLiteral(
+                        types.getPrimitiveType(TypeKind.LONG), expr.asLongLiteralExpr().asLong());
+            } else if (expr.isDoubleLiteralExpr() && !context.parsingMember) {
+                return new ValueLiteral(
+                        types.getPrimitiveType(TypeKind.DOUBLE),
+                        expr.asDoubleLiteralExpr().asDouble());
+            } else if (expr.isStringLiteralExpr() && !context.parsingMember) {
+                TypeElement stringTypeElem =
+                        env.getElementUtils().getTypeElement("java.lang.String");
+                return new ValueLiteral(
+                        types.getDeclaredType(stringTypeElem),
+                        expr.asStringLiteralExpr().asString());
+            } else if (expr.isThisExpr() && !context.parsingMember) {
```

Comment:
`ThisExpr` can also refer to an enclosing instance. After a quick check, I am not sure if the dataflow framework can handle them. For now, I'd at least add a check that the `ThisExpr` has no type name.
(In the dataflow framework, `ThisReference` confusingly seems to be used both for supertypes and outer types, which might be a bug.)